### DISCUSSION TITLE
how can i have strict rules like mypy?

### DISCUSSION THREAD

**Discussion by Heus-Sueh** [astral-sh/ruff#13216] [Q&A] [answered]
make ruff complain that variable type annotations are missing

**Comment by tjkuson** [comment, marked as answer]
You are probably looking for the [flake8-annotations (ANN)](https://docs.astral.sh/ruff/rules/#flake8-annotations-ann) ruleset. There isn't a strict mode in Ruff per se, so you'd just have to extend-select the additional rules you would like to enable (see [`lint.extend_select`](https://docs.astral.sh/ruff/settings/#lint_extend-select)). An actual strict mode is most likely waiting on #1774 (not a maintainer myself).
